identity/app/views/upsell/upsellContainer.scala.html

```diff
@@ -40,24 +40,14 @@
     }
 }
 
-
-@prefill = @{
-    Json.toJson(Map(
-        "csrfToken" -> play.filters.csrf.CSRF.getToken.map(_.value),
-        "accountToken" -> requestHeader.cookies.get("SC_GU_GUEST_PW_SET").map(_.value),
-        "email" -> Option(email),
-        "hasPassword" -> Option(hasPassword.toString),
-        "hasSocialLinks" -> Option(hasSocialLinks.toString)
-    ).collect{
-        case (name, Some(value)) => name -> value
-    }
-    ).toString()
-}
-
-<div class="js-identity-block-list" data-page-variant="@{pageVariant.jsName}">
-    <script type="text/json" data-for-prefill="true">
-        @Html(prefill)
-    </script>
+<div class="js-identity-block-list"
+    data-page-variant="@{pageVariant.jsName}"
+    data-csrf-token="@{play.filters.csrf.CSRF.getToken.fold("")(_.value)}"
+    data-account-token="@{requestHeader.cookies.get("SC_GU_GUEST_PW_SET").fold("")(_.value)}"
```

Comment:
this cookie is set as an Http only cookie, yet we inject it into the page. Would it be better to inject a flag indicating its presence, but not its actual value?

Reply:
would it be possible to pick it 100% from javascript? we have a cookie library in dotcom

Reply:
I don't think so, as `SC_GU_GUEST_PW_SET` is an HttpOnly cookie.